common/lib/dependabot/config/file.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Config
+    class File
+      attr_reader :updates, :registries
+
+      def initialize(updates:, registries: nil)
+        @updates = updates || []
+        @registries = registries || []
+      end
+
+      def update_config(package_manager, directory: nil)
+        dir = directory || "/"
+        package_ecosystem = PACKAGE_MANAGER_LOOKUP.invert.fetch(package_manager)
+        cfg = updates.find { |u| u[:"package-ecosystem"] == package_ecosystem && u[:directory] == dir }
```

Comment:
Thoughts on something like?

```suggestion
        cfg = updates.find { |u| u[:"package-ecosystem"] == package_ecosystem && u[:directory] == dir }
        raise InvalidConfigError, "no update config found for package-ecosystem #{package_ecosystem} and directory: #{dir}" unless cfg
```

Reply:
I wasn't keen on that, and [explicitly test for the opposite](https://github.com/dependabot/dependabot-core/pull/3512/files#diff-3a8a97257f0436102272ecec2896ea6f42c8794accda9c7cc65cd2b07ed46d24R37).

My thinking is:
* I may want to test (e.g. `bin/dry-run.rb`) a repository that doesn't currently have Dependabot Version Updates enabled - that should be OK.
* In a hypothetical future where this infrastructure applies to Dependabot Security Updates, an `updates:` configuration may not exist.

I don't feel super strongly, but the idea was we should return default values instead of throwing an exception.

Reply:
Ah yes of course that's a great shout, didn't think of the security update case where you don't have a config file 👌